app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,287 @@
+    private TreeMap<MessageKey, Message> mMessagesList;
+
+
+    public SnackBar(ViewGroup parent) {
+        mTargetParent = parent;
+        mContext = parent.getContext();
+
+        mMessagesList= new TreeMap<>();
+        children = new ArrayList();
+        buildList();
+    }
+
+    public void show() {
+        // TODO: discover why does not animate on the first time
+        if (mList.getVisibility() != View.VISIBLE) {
+            animate(View.VISIBLE);
+        }
+    }
+
+    public void hide() {
+        animate(View.GONE);
+    }
+
+    private void animate(int visibility) {
+        TranslateAnimation animate = new TranslateAnimation(0, 0,
+            visibility == View.VISIBLE ? mList.getHeight() : 0,
+            visibility == View.GONE ? mList.getHeight() : 0
+        );
+        animate.setDuration(700);
+        animate.setFillAfter(true);
+        mList.startAnimation(animate);
+        mList.setVisibility(visibility);
+    }
+
+    public int message(String message) {
+        return message(message, null, null, 999);
+    }
+
+    public int message(String message, int priority) {
+        return message(message, null, null, priority);
+    }
+
+    public int message(String message, String actionMessage, View.OnClickListener actionOnClick,
+                        int priority){
+        MessageKey key = new MessageKey(++mMessageId, priority);
+        Message value = new Message(key, message, actionMessage, actionOnClick);
+        mMessagesList.put(key, value);
+        updateList();
+        if (mMessagesList.size() == 0) {
+            hide();
+        } else {
+            show();
+        }
+        return mMessageId;
+    }
+
+    public void dismiss(MessageKey key) {
+        mMessagesList.remove(key);
+        updateList();
+    }
+
+    private void updateList() {
+        groups.clear();
+        children.clear();
+        children.add(new ArrayList());
+        for(Map.Entry<MessageKey, Message> entry : mMessagesList.entrySet()) {
+            Message value = entry.getValue();
+
+            if (groups.size() == 0) {
+                addToHashMap(value, groups);
+            } else {
+                addToHashMap(value, (ArrayList) children.get(0));
+            }
+        }
+        adapter.notifyDataSetChanged();
+    }
+
+    private void addToHashMap(Message m, ArrayList a) {
+        HashMap newMessage = new HashMap();
+        newMessage.put("id", Integer.toString(m.key.id));
+        newMessage.put("priority", Integer.toString(m.key.priority));
+        newMessage.put("messages", m.message);
+        if (m.actionString != null) {
+            newMessage.put("actions", m.actionString);
+        }
+        a.add(newMessage);
+    }
+
+    private void buildList() {
+        mList = new ExpandableListView(mContext);
+        setListAppearance();
+        mList.setAdapter(generateAdapter());
+        mTargetParent.addView(mList);
+        hide();
+    }
+
+    private void setListAppearance() {
```

Comment:
My initial goal was to make the whole Snackbar using only code. The only thing here that's related to dimensions is related to indicator but I thing it's ok to leave it on the code since we aren't giving this options to the user to define where it should be.